File: leetcode.js/二叉树的中序遍历.js
var inorderTraversal = function(root) {
    let result = [];
    function helper(node){
        if(!node){
            return;
        }
        helper(node.left);
        result.push(node.val);
        helper(node.right);
    }
    helper(root);
    return result;
};
// 示例使用代码
class TreeNode {
    constructor(val = 0, left = null, right = null) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

// 构建示例二叉树
//       1
//        \
//         2
//        /
//       3
const root = new TreeNode(1);
root.right = new TreeNode(2);
root.right.left = new TreeNode(3);

// 调用中序遍历函数
const result = inorderTraversal(root);
console.log(result); // 输出: [1, 3, 2]

File: leetcode.py/二叉树的中序遍历.py
class Solution(object):
    def inorderTraversal(self, root):
        result = []
        def helper(node):
            if not node:
                return
            helper(node.left)
            result.append(node.val)
            helper(node.right)

        helper(root)
        return result
# 定义二叉树节点类
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# 定义 Solution 类
class Solution(object):
    def inorderTraversal(self, root):
        result = []
        def helper(node):
            if not node:
                return
            helper(node.left)
            result.append(node.val)
            helper(node.right)

        helper(root)
        return result

# 示例使用
if __name__ == "__main__":
    # 构建二叉树
    #       1
    #        \
    #         2
    #        /
    #       3
    root = TreeNode(1)
    root.right = TreeNode(2)
    root.right.left = TreeNode(3)

    solution = Solution()
    print(solution.inorderTraversal(root))  # 输出: [1, 3, 2]
